### PR TITLE
Gql contentful bb

### PR DESCRIPTION
**Pull request by asmab89**

### CODE REVIEW

File: core/src/main/kotlin/org/evomaster/core/problem/graphql/builder/StateBuilder.kt [hunk 1]
```diff
@@ -18,6 +19,11 @@ object StateBuilder {
         "__Field", "__InputValue", "__Type", "__TypeKind"
     )
 
+    /**
+     * Used to create unique IDs
+     */
+    private val idGenerator = AtomicInteger()
```
Comment (arcuri82) [inline core/src/main/kotlin/org/evomaster/core/problem/graphql/builder/StateBuilder.kt:25]: this must be avoided, ie mutable static state. 
if really need it as static state, rather use `StaticCounter`

File: core/src/main/kotlin/org/evomaster/core/problem/graphql/builder/TempState.kt [hunk 2]
```diff
@@ -25,5 +25,7 @@ data class TempState(
         /*
          * An intermediate data structure used for extracting Union types
          */
-        var tempUnionTables: MutableList<Table> = mutableListOf()
+        var tempUnionTables: MutableList<Table> = mutableListOf(),
+
+        var inputTypeName: MutableMap<String,String> = mutableMapOf()
```
Comment (arcuri82) [inline core/src/main/kotlin/org/evomaster/core/problem/graphql/builder/TempState.kt:30]: what are the `key` and `value` here in this map? add documentation

File: core/src/main/kotlin/org/evomaster/core/problem/graphql/builder/StateBuilder.kt (hunk 1 above)
Comment (asmab89) [reply core/src/main/kotlin/org/evomaster/core/problem/graphql/builder/StateBuilder.kt]: Done.

File: core/src/main/kotlin/org/evomaster/core/problem/graphql/builder/TempState.kt (hunk 2 above)
Comment (asmab89) [reply core/src/main/kotlin/org/evomaster/core/problem/graphql/builder/TempState.kt:35]: Done.

**Pull request merged**